packages/core/src/providers/apple.ts

```diff
@@ -158,17 +170,38 @@ export default function Apple<P extends AppleProfile>(
     name: "Apple",
     type: "oidc",
     issuer: "https://appleid.apple.com",
+    wellKnown: "https://appleid.apple.com/.well-known/openid-configuration",
+    checks: ["pkce"],
+    token: {
+      url: `https://appleid.apple.com/auth/token`,
+    },
     authorization: {
-      params: { scope: "name email", response_mode: "form_post" },
+      url: 'https://appleid.apple.com/auth/authorize',
+      params: {
+        scope: "name email",
+        response_type: "code",
+        response_mode: "form_post",
+        state: crypto.randomUUID()
+      },
```

Comment:
```suggestion
      params: {
        scope: "name email",
        response_mode: "form_post",
      },
```

Already added/default.